classes/local/store/s3/client.php

```diff
@@ -422,6 +428,21 @@ public function support_presigned_urls() {
         return true;
     }
 
+    /**
+     * Returns domain bit for redirect URL.
+     *
+     * @return string
+     */
+    public function presigned_domain() {
+        if ('cf' === $this->signingmethod) {
+            return $this->cloudfrontresourcedomain;
+        }
+
+        $key = 'fake-key';
+        $domain = $this->client->getObjectUrl($this->bucket, $key);
```

Comment:
does this call have any performance impact? ie does it do any network stuff or is it purely client side?

Reply:
Documentation says "Builds an URL to access object. Does not check if object exists, ACL etc". Maybe not exact words. So I take it does not do any S3 requests. I'll check anyway.